src/main/java/org/apache/maven/plugins/dependency/filters/AbstractDependencyFilter.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.maven.plugins.dependency.filters;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Set;
+
+
+abstract class AbstractDependencyFilter implements DependencyFilter
```

Comment:
Hello @bmarwell ,
	Since you created this abstract class, maybe it could contain more than just `splitValues` method. The inherited classes are very similar IMO; This way, this abstract class could have almost all logic here, with the filter, include and exclude methods; It is just a suggestion to avoid code duplication.

Below an example of the abstract class and one of its child. Even the `filterexcludeIds` and `filterincludeIds` could be simplified because they share similar logic.

As ScopeFilter has specific treatment to split include and exclue ids, I created two functions `splitIncludeIds` and `splitExcludeIds`

```java
	abstract class AbstractDependencyFilter implements DependencyFilter
	{
	protected final String includeIds;

	protected final String excludeIds;

	public AbstractDependencyFilter(String includeIds, String excludeIds)
	{
		this.includeIds = includeIds == null ? "" : includeIds;
		this.excludeIds = excludeIds == null ? "" : excludeIds;
	}


	@Override
	public Set<Dependency> filter(Set<Dependency> dependencies)
	{
		Set<Dependency> filtered = new HashSet<>(dependencies);

		filtered = filterincludeIds(filtered);
		filtered = filterexcludeIds(filtered);

		return filtered;
	}

	private Set<Dependency> filterexcludeIds(Set<Dependency> dependencies)
	{
		if (excludeIds.trim().isEmpty()) {
			return dependencies;
		}

		final Set<String> excludedIds = splitExcludeIds(excludeIds);

		Set<Dependency> filtered = new HashSet<>(dependencies.size());
		for (Dependency dependency : dependencies) {
			if (excludedIds.contains(getContainsProperty(dependency))) {
				continue;
			}

			filtered.add(dependency);
		}

		return filtered;
	}


	private Set<Dependency> filterincludeIds(Set<Dependency> dependencies)
	{
		if (includeIds.trim().isEmpty()) {
			return dependencies;
		}

		Set<String> includedIds = splitIncludeIds(includeIds);

		Set<Dependency> filtered = new HashSet<>(dependencies.size());
		for (Dependency dependency : dependencies) {
			if (includedIds.contains(getContainsProperty(dependency))) {
				filtered.add(dependency);
			}
		}

		return filtered;
	}

	protected Set<String> splitExcludeIds(String includeIds) {
		return splitValues(includeIds);
	}

	protected Set<String> splitIncludeIds(String includeIds) {
		return splitValues(includeIds);
	}

	protected Set<String> splitValues(String excludeIds) {
		// ommited here
	}

	abstract protected String getContainsProperty(Dependency dependency);

}
```

```java
	public class GroupIdFilter extends AbstractDependencyFilter
	{

		public GroupIdFilter(String includeGroupIds, String excludeGroupIds)
		{
			super(includeGroupIds, excludeGroupIds);
		}

	    protected String getContainsProperty(Dependency dependency) {
			return dependency.getGroupId();
		}

	}
```

Reply:
I was thinking of closing this PR. We would need the same logic for other goals as well, and this goal could be replaced with the combination of two other goals.

I think you could just create a PR to my branch of you're interested to keep it up.

Reply:
Created the PR to your branch: https://github.com/apache/maven-dependency-plugin/pull/145

Should I change anything else? There is a apparently unfinished thread about package names in June 1st. 

Do I have to sign any documents?  Please let me know.

PS.: I am sorry, butI did not understand why you are thinking in close this PR. Which goals are you talking about? Should they be a part in this PR? I think this PR solves the go-offline problem, which is the subject of MDEP-568.